add_trailing_comma/_plugins/pep695.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+import ast
+import sys
+from typing import Iterable
+
+from tokenize_rt import Offset
+from tokenize_rt import Token
+
+from add_trailing_comma._ast_helpers import ast_to_offset
+from add_trailing_comma._data import register
+from add_trailing_comma._data import State
+from add_trailing_comma._data import TokenFunc
+from add_trailing_comma._token_helpers import find_simple
+from add_trailing_comma._token_helpers import fix_brace
+
+
+if sys.version_info >= (3, 12):  # pragma: >=3.12 cover
+    def _fix_pep695(
+        i: int,
+        tokens: list[Token],
+    ) -> None:
+        for n in range(i, len(tokens)):
+            token = tokens[n]
+            if token.name == 'OP' and token.src == '[':
+                return fix_brace(
+                    tokens,
+                    find_simple(n, tokens),
+                    add_comma=True,
+                    remove_comma=True,
+                )
+        else:
+            raise AssertionError('Past end?')
+
+    @register(ast.TypeAlias)
+    @register(ast.ClassDef)
+    @register(ast.FunctionDef)
+    def visit_ast(
+        state: State,
+        node: ast.TypeAlias | ast.ClassDef | ast.FunctionDef,
+    ) -> Iterable[tuple[Offset, TokenFunc]]:
+        if node.type_params:
+            yield ast_to_offset(node), _fix_pep695
```

Comment:
ah hmmm mypy isn't type checking this because it's hidden behind a 3.12 block -- but will error when run with a 3.12 mypy (the way the decorator is written it can only register a single concrete type at once rather than a union to prevent mistakes) -- so this'll have to be 3 separate functions that `yield from _common_function(...)`

Reply:
or I guess do what `functions.py` does -- calling the decorator directly seems to work nicely